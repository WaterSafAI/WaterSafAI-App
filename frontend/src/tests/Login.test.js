import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Login from '../screens/Login';

// Mock the useAuth hook
const mockLogin = jest.fn();
jest.mock("../services/AuthProvider", () => ({
  useAuth: () => ({
    actions: {
      login: mockLogin,
      // ... other actions
    },
    loading: false,
  }),
}));

describe('Login Component', () => {
  const mockNavigation = { push: jest.fn() };

  it('renders correctly', () => {
    const tree = render(<Login navigation={mockNavigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('allows entering email and password', () => {
    const { getByPlaceholderText } = render(<Login navigation={mockNavigation} />);

    fireEvent.changeText(getByPlaceholderText('Email'), 'test@example.com');
    fireEvent.changeText(getByPlaceholderText('Password'), 'password');
    
    // Additional expectations can be added here if needed
  });

  it('navigates to Register screen on button press', () => {
    const { getByText } = render(<Login navigation={mockNavigation} />);
    
    fireEvent.press(getByText('Sign Up'));
    expect(mockNavigation.push).toHaveBeenCalledWith("Register");
  });

  it('calls the forgot password function on button press', () => {
    const { getByText } = render(<Login navigation={mockNavigation} />);
    fireEvent.press(getByText('Forgot Password?'));
    // Expect some action to be called, like a console.log or navigation
  });

  it('calls the login function with correct credentials on button press', () => {
    const { getByPlaceholderText, getByText } = render(<Login navigation={mockNavigation} />);
    const emailInput = getByPlaceholderText('Email');
    const passwordInput = getByPlaceholderText('Password');
    const loginButton = getByText('LOG IN');

    const testEmail = 'test@example.com';
    const testPassword = 'password123';

    fireEvent.changeText(emailInput, testEmail);
    fireEvent.changeText(passwordInput, testPassword);
    fireEvent.press(loginButton);

    // Expect the mockLogin function to have been called with correct credentials
    expect(mockLogin).toHaveBeenCalledWith(testEmail, testPassword);
  });
});
