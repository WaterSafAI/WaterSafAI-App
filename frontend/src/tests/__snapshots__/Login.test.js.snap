// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ADD8E6",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "flex-start",
    }
  }
>
  <Text
    style={
      {
        "color": "#0A3465",
        "fontSize": 32,
        "marginTop": 50,
      }
    }
  >
    WaterSaf
    <Text
      style={
        {
          "fontWeight": "bold",
        }
      }
    >
      AI
    </Text>
  </Text>
  <Image
    source={
      {
        "testUri": "../../../src/assets/logo-img.png",
      }
    }
    style={
      {
        "height": 150,
        "marginBottom": 20,
        "width": 150,
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "#D8EBF1",
        "borderColor": "#644535",
        "borderRadius": 15,
        "borderWidth": 3,
        "height": 60,
        "justifyContent": "center",
        "margin": 10,
        "padding": 10,
        "width": "80%",
      }
    }
  >
    <TextInput
      fontSize={18}
      onChangeText={[Function]}
      placeholder="Email"
      placeholderTextColor="#644535"
      style={
        {
          "height": 60,
        }
      }
    />
  </View>
  <View
    style={
      {
        "backgroundColor": "#D8EBF1",
        "borderColor": "#644535",
        "borderRadius": 15,
        "borderWidth": 3,
        "height": 60,
        "justifyContent": "center",
        "margin": 10,
        "padding": 10,
        "width": "80%",
      }
    }
  >
    <TextInput
      fontSize={18}
      onChangeText={[Function]}
      placeholder="Password"
      placeholderTextColor="#644535"
      secureTextEntry={true}
      style={
        {
          "height": 60,
        }
      }
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "margin": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 15,
          "fontWeight": "bold",
          "textDecorationLine": "underline",
          "textDecorationStyle": "solid",
        }
      }
    >
      Forgot Password?
    </Text>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "backgroundColor": "#66B3CC",
        },
        {
          "borderRadius": 15,
          "borderWidth": 3,
          "height": 60,
          "justifyContent": "center",
          "padding": 15,
          "width": "80%",
        },
      ]
    }
  >
    <Text
      style={
        {
          "color": "#FFFFFF",
          "fontSize": 22,
          "fontWeight": "bold",
          "textAlign": "center",
        }
      }
    >
      LOG IN
    </Text>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "margin": 20,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      Don't have an account?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "marginLeft": 4,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 15,
            "fontWeight": "bold",
            "textDecorationLine": "underline",
            "textDecorationStyle": "solid",
          }
        }
      >
        Sign Up
      </Text>
    </View>
  </View>
</View>
`;
